src/pandablocks_ioc/ioc.py

```diff
@@ -874,6 +925,28 @@ def _make_pos_out(
             PviGroup.CAPTURE,
             labels=labels,
             initial_value=capture_index,
+            on_update=lambda new_capture_mode: (
+                self._dataset_name_cache.update_cache(
+                    record_name,
+                    record_dict[dataset_record_name].record.get(),
+                    labels[new_capture_mode],
+                )
+            ),
```

Comment:
This overrides the `on_update` with something that updates the cache, but we still need the old behaviour of actually sending the capture value to the PandA. Maybe https://github.com/PandABlocks/PandABlocks-ioc/blob/8e25d64cc97cfbf242a84c6ccbe600a39ed41c41/src/pandablocks_ioc/ioc.py#L664-L676 needs updating so that it calls the user supplied `on_update` *and* sends the value to PandA? I haven't checked where else we call `_create_record_info` though...

Reply:
I'm confused what you mean here? This code doesn't currently have an `on_update`?

``` python
        record_dict[capture_record_name] = self._create_record_info(
            capture_record_name,
            "Capture options",
            builder.mbbOut,
            int,
            PviGroup.CAPTURE,
            labels=labels,
            initial_value=capture_index,
        )
```

Reply:
Inside `_create_record_info` there is a check for whether `on_update` has been passed in as a kwarg. Looking at the diff viewer there definitely is a `on_update` currently checked in for the `capture_record_name` record.

Reply:
I think this should be fine, `_create_record_info` is only in ioc.py anyway

Reply:
> Inside `_create_record_info` there is a check for whether `on_update` has been passed in as a kwarg. Looking at the diff viewer there definitely is a `on_update` currently checked in for the `capture_record_name` record.

I'm confused, I can't see any mention of `on_update` here:

https://github.com/PandABlocks/PandABlocks-ioc/blob/8e25d64cc97cfbf242a84c6ccbe600a39ed41c41/src/pandablocks_ioc/ioc.py#L865-L877

so that means it will prod the `record_updater` on `caput` which I assume pushes a value to PandA?

In the change here then a different `on_update` is supplied so no `record_updater` ever is made so no value is pushed to the PandA?

Reply:
I did a quick fix by creating the record updater in `_make_pos_out` `_make_ext_out` - the alternatives would leave me with annoying errors.

I made a new issue for solving the problem more generally https://github.com/PandABlocks/PandABlocks-ioc/issues/121